adm_program/system/classes/User.php

```diff
@@ -1119,93 +1119,109 @@ public function getValue(string $columnName, string $format = '')
+            $vCard[] = 'X-SOCIALPROFILE;TYPE=MASTODON:' . $this->getValue('MASTODON');
+        }
+
+        if ($gCurrentUser->allowedViewProfileField($this, 'BIRTHDAY') && $this->getValue('BIRTHDAY') !== '') {
+            $vCard[] = 'BDAY:' . $this->getValue('BIRTHDAY', 'Y-m-d');
         }
-        if ($gCurrentUser->allowedViewProfileField($this, 'EMAIL')) {
-            $vCard[] = 'EMAIL;PREF;INTERNET:' . $this->getValue('EMAIL');
+        if ($gCurrentUser->allowedViewProfileField($this, 'EMAIL') && $this->getValue('EMAIL') !== '') {
+            $vCard[] = 'EMAIL;TYPE=home:' . $this->getValue('EMAIL');
         }
         $file = ADMIDIO_PATH . FOLDER_DATA . '/user_profile_photos/' . (int)$this->getValue('usr_id') . '.jpg';
         if ((int)$gSettingsManager->get('profile_photo_storage') === 1 && is_file($file)) {
             $imgHandle = fopen($file, 'rb');
             if ($imgHandle !== false) {
                 $base64Image = base64_encode(fread($imgHandle, filesize($file)));
                 fclose($imgHandle);
-
-                $vCard[] = 'PHOTO;ENCODING=BASE64;TYPE=JPEG:' . $base64Image;
+                $vCard[] = 'PHOTO;TYPE=JPEG;ENCODING=b:' . $base64Image;
             }
         }
-        if ((int)$gSettingsManager->get('profile_photo_storage') === 0 && $this->getValue('usr_photo') !== '') {
-            $vCard[] = 'PHOTO;ENCODING=BASE64;TYPE=JPEG:' . base64_encode((string) $this->getValue('usr_photo'));
+        if ((int)$gSettingsManager->get('profile_photo_storage') === 0 && !is_null($this->getValue('usr_photo'))) {
+            $vCard[] = 'PHOTO;TYPE=JPEG;ENCODING=b:' . base64_encode((string) $this->getValue('usr_photo'));
         }
-        // Gender is not included in vCard 2.1, is passed here for the Windows address book
-        if ($gCurrentUser->allowedViewProfileField($this, 'GENDER') && $this->getValue('GENDER') > 0) {
+        if ($gCurrentUser->allowedViewProfileField($this, 'GENDER') && (int)$this->getValue('GENDER') > 0) {
+            // https://datatracker.ietf.org/doc/html/rfc6350#section-6.2.7
             if ((int)$this->getValue('GENDER') === 1) {
-                $xGender = 'Male';
-                $xWabGender = 2;
-            } else {
-                $xGender = 'Female';
-                $xWabGender = 1;
+                $vCard[] = 'GENDER:M';
+            } elseif ((int)$this->getValue('GENDER') === 2) {
+                $vCard[] = 'GENDER:F';
+            } elseif ((int)$this->getValue('GENDER') === 3) {
+                $vCard[] = 'GENDER:O';
             }
-
-            $vCard[] = 'X-GENDER:' . $xGender;
-            $vCard[] = 'X-WAB-GENDER:' . $xWabGender;
         }
         if ($this->getValue('usr_timestamp_change') !== '') {
-            $vCard[] = 'REV:' . $this->getValue('usr_timestamp_change', 'Ymd\This');
+            $vCard[] = 'REV:' . $this->getValue('usr_timestamp_change', 'c');
+        }
+        if ($this->getValue('usr_uuid') !== '') {
```

Comment:
Every user record should have a usr_uuid so you don't need to check if it's empty or not.

Reply:
thanks, I will fix that